### ISSUE TITLE
Licensing of some basic modules

### ISSUE THREAD

**Issue by blaggacao** [OCA/partner-contact#321]
I guess there is very interesting and valuable, and above all reusable work done in this repository, however it seems that for a licensing conflict we cannot use it for localisation modules. First it needs to be acknowledge, that we can only publish under LGPL in order not to automatically produce kind of a license conflict with our more ample private developments. But then, on the other hand we would like to incorporate some modules of this repository (and possibly others) in order to honour the idea of reusability and shared effort (to avoid "reinventing the wheel"). 
Therefore, I would like to ask very humbly, if there could be any chance of relicensing some key common modules under a LGPL license? I can understand why this can be a bad idea, but unfortunately this automatically means, that no LGPL licensed module can use them, which inhibits wider adoption (among honourable and license-respecting people, which cannot adopt AGPL for whatever multiple reasons, like company policy, externally licensed mixins, etc.)

**Comment by hbrunn** [comment]
for any of my commits, it's a non-negotiable no

**Comment by blaggacao** [comment]
I see that the following modules could be of special interest:
- `base_location` - as it is almost inevitable that more advanced modules would depend on some kind of data produced under this facilities. (read=city based conditionals)
- `base_location_geonames_import` - the previous argument goes unchanged here
- `partner_identification` - this is a great base for amplification and adaption and make it a core component of private logic all around any possible imaginable ID.

**Comment by yajo** [comment]
As long as any of you want to sue bad uses of your AGPL-licensed addons at any point in the future, it's rational to keep that license.

If you just license it in AGPL but either use it as a dependency of an LGPL/private addon (illegal) or don't want to take any action on anybody doing that, then just license it under LGPL and forget about legal issues. It's open source too, after all.

OCA's standing point is more the 2nd point, so we should either change that position or relicense to LGPL.

Of course, you must respect every author before doing that on any addon. You could open a relicense PR for each of those and ping all original authors and wait for their input, but I'm pretty sure many will not agree...

In fact, the most pragmatic approach would be you licensing all under AGPL and continuing under that license.

I'm sure I just started a flamewar, but I don't want to, so I'm closing this. Choose your option and do it:
1. Relicense yours to AGPL.
2. Open relicense PRs and ping authors.

**Comment by blaggacao** [comment]
@Yajo can you clarify:

> OCA's standing point is more the 2nd point, so we should either change that position or relicense to LGPL.
> From the previous paragraphs I understand:

Point 1: AGPL is good if you wants to keep the option to sue bad license use, for example as a strategic option in your market.
Point 2: If you can, license LGPL, and forget about the issues downstream.

OCA position: Point 2? I cannot see how changing this is the alternative to relicensing.

In my case, if I want to include upstream work, my work is getting infected. On the other hand, I cannot use AGPL, because a **third party** would be happy to extract our dependent IP through legal action, then.

**Comment by yajo** [comment]
Point 1:

> you want to sue bad uses of your AGPL-licensed addons 

Point 2 (OCA's current position):

> If you just license it in AGPL but [...] don't want to take any action on anybody doing that

In your case, if you depend on AGPL work, you can only license yours under AGPL.

> I cannot use AGPL, because a third party would be happy to extract our dependent IP through legal action, then.

I don't understand that sentence. What's IP?

**Comment by blaggacao** [comment]
IP = Intellectual Property - This is possible because OCA position is just that: an opinion. Legally binding is the license and this opens the door to a competitor, who does not maintain OCA's position to extract the intellectual property forcibly. Of course, anyone is entitled to sue for AGPL compliance, after all it's a public license which public interest, so if the interest of "the public" is at stake, "the public" can take legal action.

**Comment by rvalyi** [comment]
@blaggacao I would rather re-phrase "I cannot use AGPL, because a third party would be happy to extract our dependent IP through legal action, then" as "because third parties prefer putting some pressure so that businesses using their code forward some code back to the ecosystem by playing by the same rules."

**Comment by yajo** [comment]
But I don't understand what you mean. In fact, publishing under AGPL gives you more rights and protects more your code. You can still publish your assets under a different license, because assets are not code.

**Comment by blaggacao** [comment]
@rvalyi I don't believe the benefits of feeding back code exceed the risks incurring in extra expense by having legal action taken against me. Or maybe I didn't understand your argument well...

@Yajo AGPL comes with rights AND obligations. Enforcing my rights is on my pocket. Obeying the obligations is matter of honour and good conduct it doesn't cost anyone anything in my case. The obligation I incur is furthermore not only costly, but inacceptable if I want to be able to cross license down the line in the future.

In conclusion: choosing AGPL, I can screw myself, as it is more restrictive. The effect is even worse, if I don't have the means to effectively enforce my rights properly. Which is almost always the case, in practice, I guess.
